src/icloudpd/base.py

```diff
@@ -1360,6 +1373,25 @@ def should_break(counter: Counter) -> bool:
             )
             photos_counter = 0
 
+            logger.debug(f"Keep iCloud album: {keep_icloud_album}")
+            try:
+                keep_icloud_album_obj = (
+                    library_object.albums[keep_icloud_album] if keep_icloud_album else None
+                )
+                keep_icloud_album_photo_ids = (
+                    {photo.id for photo in keep_icloud_album_obj.photos}
+                    if keep_icloud_album_obj
+                    else set()
+                )
+            except KeyError:
+                if keep_icloud_album:
```

Comment:
If user put wrong album name we warn about it and still delete all photos from it. Better user experience would be to check for album and stop action. Probably better to check before we start iteration on photos, so in case `--watch-with-interval` is used, we check for album every interval (e.g. if album was removed in iCloud).

Reply:
Updated this to return early